### PR TITLE
Cpp17

### PR DESCRIPTION
**Pull request by FG-TUM**
# Description

bumped c++ version from 14 to 17

## ~Resolved Issues # (issue)~

# How Has This Been Tested?

see Jenkins results
- [ ] tested mardyn integration

### CODE REVIEW

**Comment by FG-TUM** [comment]
@SteffenSeckler The CUDA compiler Jenkins uses does not support cuda + c++17

**Comment by SteffenSeckler** [comment]
well as clang is currently not properly supported in cmake as a cuda compiler, we will probably have to hold off with c++17.
See also: https://gitlab.kitware.com/cmake/cmake/issues/16586

File: src/autopas/CMakeLists.txt [hunk 1]
```diff
@@ -13,6 +13,8 @@ ADD_LIBRARY(autopas STATIC
         $<$<BOOL:${ENABLE_CUDA}>:${CU_SRC}>
         )
 
+target_compile_features(autopas PRIVATE cxx_std_17)
```
Comment (SteffenSeckler) [inline src/autopas/CMakeLists.txt]: should definitely NOT be private!

Comment (SteffenSeckler) [reply src/autopas/CMakeLists.txt]: but public

**Comment by FG-TUM** [comment]
Closed because currently not fixable

**Pull request closed without merging**